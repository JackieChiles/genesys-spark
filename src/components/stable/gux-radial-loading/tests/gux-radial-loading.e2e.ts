import { newE2EPage } from '@stencil/core/testing';

describe('gux-radial-loading', () => {
  [
    '<gux-radial-loading></gux-radial-loading>',
    '<gux-radial-loading context="modal"></gux-radial-loading>',
    '<gux-radial-loading context="input"></gux-radial-loading>',
    '<gux-radial-loading context="full-page"></gux-radial-loading>'
  ].forEach((content, index) => {
    it(`should display component as expected (${index + 1})`, async () => {
      const page = await newE2EPage();

      await page.setContent(content);

      const element = await page.find('gux-radial-loading');

      expect(element.innerHTML).toMatchSnapshot();
    });
  });

  it('should add an aria-label with the provided screenreader-text to the progressbar', async () => {
    const page = await newE2EPage();

    await page.setContent(
      '<gux-radial-loading screenreader-text="Loading Content"></gux-radial-loading>'
    );

    const progressbar = await page.find(
      'gux-radial-loading div[role="progressbar"]'
    );

    expect(progressbar.getAttribute('aria-label')).toEqual('Loading Content');
  });

  it('should add the default aria-label text if no screenreader-text is provided', async () => {
    const page = await newE2EPage();

    await page.setContent('<gux-radial-loading></gux-radial-loading>');

    const progressbar = await page.find(
      'gux-radial-loading div[role="progressbar"]'
    );

    expect(progressbar.getAttribute('aria-label')).toEqual('Loading');
  });
});
