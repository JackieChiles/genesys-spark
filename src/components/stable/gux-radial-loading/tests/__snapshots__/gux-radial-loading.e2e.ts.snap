// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-radial-loading should display component as expected (1) 1`] = `"<div role=\\"progressbar\\" aria-label=\\"Loading\\" class=\\"gux-spinner-container gux-modal\\"><div role=\\"presentation\\" class=\\"gux-spin-circle\\"></div></div>"`;

exports[`gux-radial-loading should display component as expected (2) 1`] = `"<div role=\\"progressbar\\" aria-label=\\"Loading\\" class=\\"gux-spinner-container gux-modal\\"><div role=\\"presentation\\" class=\\"gux-spin-circle\\"></div></div>"`;

exports[`gux-radial-loading should display component as expected (3) 1`] = `"<div role=\\"progressbar\\" aria-label=\\"Loading\\" class=\\"gux-spinner-container gux-input\\"><div role=\\"presentation\\" class=\\"gux-spin-circle\\"></div></div>"`;

exports[`gux-radial-loading should display component as expected (4) 1`] = `"<div role=\\"progressbar\\" aria-label=\\"Loading\\" class=\\"gux-spinner-container gux-full-page\\"><div role=\\"presentation\\" class=\\"gux-spin-circle\\"></div></div>"`;
